tests/Requests/MiddlewareTest.php

```diff
@@ -5,30 +5,68 @@
 namespace OpenEuropa\EPoetry\Tests\Requests;
 
 use GuzzleHttp\Psr7\Response;
+use OpenEuropa\EPoetry\Middleware\CasProxyTicketSessionMiddleware;
 use OpenEuropa\EPoetry\Tests\Middleware\MockMiddleware;
 use OpenEuropa\EPoetry\Type\CreateRequests;
-use OpenEuropa\EPoetry\Type\LinguisticRequestIn;
-use OpenEuropa\EPoetry\Type\RequestGeneralInfoIn;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
 
 /**
  * @internal
  * @coversNothing
  */
 final class MiddlewareTest extends AbstractTest
 {
+    /**
+     * Test exception for missing Proxy Ticket in request.
+     */
+    public function testNoProxyTicketException()
+    {
+        // Generate response.
+        $content = file_get_contents(self::FIXTURE_DIR . '/create-requests-response.xml');
+        $response = new Response(200, [], $content);
+        $this->httpClient->addResponse($response);
+
+        $clientFactory = $this->createClientFactory();
+        $session = new Session(new MockArraySessionStorage());
+        $middleware = new CasProxyTicketSessionMiddleware($session);
+        $clientFactory->addMiddleware($middleware);
+        $client = $clientFactory->getClient();
+
+        // Try to perform the request but expect an exception.
+        $this->expectExceptionMessage('[epoetry] session has no proxy ticket.');
+        $client->createRequests(new CreateRequests());
```

Comment:
Maybe we could turn this test into one that tests that logger logs the missing ticket correctly in case we wat to go down that road?

Reply:
For me it's really an exception... which should be logged, and we can test that. But I would wait OPENEUROPA-1551 to be merged.

Reply:
I would rather inject the logger in the middleware in a followup and then have the middleware log it.